Prevent unnecessary Touch ID prompt on macOS in the remember-unlock branch

sebastianlivoni opened the pull request:
[NOTE]: # ( Describe your changes in detail. Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX". )
[NOTE]: # ( If you used Generative AI to write the majority of your code, you must state this. )

On macOS, it would display a Touch ID prompt every time KeePassXC was opened because the hasKey check triggered authentication UI. It now sets `kSecUseAuthenticationUI` to `kSecUseAuthenticationUIFail` using `CFDictionarySetValue` to check whether the item exists in the keychain without invoking authentication. If status == errSecInteractionNotAllowed, the item exist.

It now also uses LAContext because kSecUseOperationPrompt is deprecated.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change that fixes an issue)

droidmonkey committed:
Add support to remember quick unlock on Windows and macOS

droidmonkey committed:
Add Pin Quick Unlock option

* Introduce QuickUnlockManager to fall back to pin unlock if OS native options are not available.

droidmonkey committed:
Add safeguard to using Argon2 function

droidmonkey committed:
Address PR comments

droidmonkey committed:
WIP: Enable centralized secret storage

* Also enables pin unlock to be stored

TODO: Clean up pin unlock interface with polkit

droidmonkey commented:
I needed to push a ton of changes to that branch, which I just did, so please rebase your commit onto the new branch head. Note that all the TouchID ObjectiveC code was moved to src/gui/osutils/macutils/AppKitImpl.mm

sebastianlivoni committed:
Fix repeating Touch ID prompt on MacOS

Also adds the APPLE_TEAM_ID and APPLE_APP_IDENTIFIER variables to CMake for using the keychain with a custom provisioning profile.

sebastianlivoni commented:
I’ve rebased the commit and made some additional updates. Specifically, I added APPLE_TEAM_ID and APPLE_APP_IDENTIFIER to the CMake configuration to use my own Apple Developer provisioning profile for keychain access.

droidmonkey commented:
I cherry picked this onto the other PR, thank you
